PWGCF/Flow/Tasks/GFWcascadeflow.cpp

```diff
@@ -0,0 +1,434 @@
+    fMultPVCutHigh->SetParameters(3257.29, -121.848, 1.98492, -0.0172128, 6.47528e-05, 154.756, -1.86072, -0.0274713, 0.000633499, -3.37757e-06);
+    fMultCutLow = new TF1("fMultCutLow", "[0]+[1]*x+[2]*x*x+[3]*x*x*x - 2.*([4]+[5]*x+[6]*x*x+[7]*x*x*x+[8]*x*x*x*x)", 0, 100);
+    fMultCutLow->SetParameters(1654.46, -47.2379, 0.449833, -0.0014125, 150.773, -3.67334, 0.0530503, -0.000614061, 3.15956e-06);
+    fMultCutHigh = new TF1("fMultCutHigh", "[0]+[1]*x+[2]*x*x+[3]*x*x*x + 3.*([4]+[5]*x+[6]*x*x+[7]*x*x*x+[8]*x*x*x*x)", 0, 100);
+    fMultCutHigh->SetParameters(1654.46, -47.2379, 0.449833, -0.0014125, 150.773, -3.67334, 0.0530503, -0.000614061, 3.15956e-06);
+    fT0AV0AMean = new TF1("fT0AV0AMean", "[0]+[1]*x", 0, 200000);
+    fT0AV0AMean->SetParameters(-1601.0581, 9.417652e-01);
+    fT0AV0ASigma = new TF1("fT0AV0ASigma", "[0]+[1]*x+[2]*x*x+[3]*x*x*x+[4]*x*x*x*x", 0, 200000);
+    fT0AV0ASigma->SetParameters(463.4144, 6.796509e-02, -9.097136e-07, 7.971088e-12, -2.600581e-17);
+  }
+
+  template <char... chars>
+  void FillProfile(const GFW::CorrConfig& corrconf, const ConstStr<chars...>& tarName, const double& cent)
+  {
+    float dnx = 0;
+    float val = 0;
+    dnx = fGFW->Calculate(corrconf, 0, kTRUE).real();
+    if (dnx == 0)
+      return;
+    if (!corrconf.pTDif) {
+      val = fGFW->Calculate(corrconf, 0, kFALSE).real() / dnx;
+      if (TMath::Abs(val) < 1) {
+        registry.fill(tarName, cent, val, dnx);
+      }
+      return;
+    }
+    return;
+  }
+  template <char... chars>
+  void FillProfilepTOmega(const GFW::CorrConfig& corrconf, const ConstStr<chars...>& tarName, const double& cent, const int& ptbin)
+  {
+    float dnx = 0;
+    float val = 0;
+    for (int massbin = 1; massbin <= nOmegaMassBins; massbin++) {
+      dnx = fGFW->Calculate(corrconf, (ptbin - 1) + ((massbin - 1) * nPtBins), kTRUE).real();
+      if (dnx == 0)
+        continue;
+      val = fGFW->Calculate(corrconf, (ptbin - 1) + ((massbin - 1) * nPtBins), kFALSE).real() / dnx;
+      if (TMath::Abs(val) < 1) {
+        registry.fill(tarName, fPtAxis->GetBinCenter(ptbin), fOmegaMass->GetBinCenter(massbin), val, dnx);
+      }
+    }
+    return;
+  }
+  template <char... chars>
+  void FillProfilepTXi(const GFW::CorrConfig& corrconf, const ConstStr<chars...>& tarName, const double& cent, const int& ptbin)
+  {
+    float dnx = 0;
+    float val = 0;
+    for (int massbin = 1; massbin <= nXiMassBins; massbin++) {
+      dnx = fGFW->Calculate(corrconf, (ptbin - 1) + ((massbin - 1) * nPtBins), kTRUE).real();
+      if (dnx == 0)
+        continue;
+      val = fGFW->Calculate(corrconf, (ptbin - 1) + ((massbin - 1) * nPtBins), kFALSE).real() / dnx;
+      if (TMath::Abs(val) < 1) {
+        registry.fill(tarName, fPtAxis->GetBinCenter(ptbin), fXiMass->GetBinCenter(massbin), val, dnx);
+      }
+    }
+    return;
+  }
```

Comment:
In here basically the code is replicated
Replicating code is error prone and difficult to maintain and evolve
The template mechanism can be further used to avoid code replication

Reply:
In fact the only different there is the massaxis. So maybe a batter change is that let the massaxis become an input, I think.